src/notation/internal/notationinteraction.cpp

```diff
@@ -2182,47 +2436,94 @@ static int findGrip(const QVector<mu::RectF>& grips, const mu::PointF& canvasPos
 
 bool NotationInteraction::isHitGrip(const PointF& pos) const
 {
-    return selection()->element() && findGrip(m_gripEditData.grip, pos) != -1;
+    return selection()->element() && findGrip(m_editData.grip, pos) != -1;
 }
 
 void NotationInteraction::startEditGrip(const PointF& pos)
 {
-    int grip = findGrip(m_gripEditData.grip, pos);
+    int grip = findGrip(m_editData.grip, pos);
     if (grip == -1) {
         return;
     }
+    startEditGrip(Ms::Grip(grip));
+}
 
-    m_gripEditData.curGrip = Ms::Grip(grip);
-
+void NotationInteraction::updateAnchorLines()
+{
     std::vector<LineF> lines;
-    QVector<LineF> anchorLines = m_gripEditData.element->gripAnchorLines(m_gripEditData.curGrip);
+    Ms::Grip anchorLinesGrip = m_editData.curGrip == Ms::Grip::NO_GRIP ? m_editData.element->defaultGrip() : m_editData.curGrip;
+    QVector<LineF> anchorLines = m_editData.element->gripAnchorLines(anchorLinesGrip);
 
-    EngravingItem* page = m_gripEditData.element->findAncestor(ElementType::PAGE);
-    const PointF pageOffset((page ? page : m_gripEditData.element)->pos());
+    EngravingItem* page = m_editData.element->findAncestor(ElementType::PAGE);
     if (!anchorLines.isEmpty()) {
         for (LineF& line : anchorLines) {
-            line.translate(pageOffset);
-            lines.push_back(line);
+            if (line.p1() != line.p2()) {
```

Comment:
anchor lines already canvas-relative, no need to adjust for page, but occasionally anchor lines are single points, which causes asserts if you try to render them